Sources/Utility/ArgumentParser.swift

```diff
@@ -447,3 +447,166 @@ public final class ArgumentParser: ArgumentParserProtocol {
         }
     }
 }
+
+/// A protocol representing an option structure, which can be
+/// bounded in the BoundParser.
+public protocol BoundParserOption {
+    init()
+}
+
+/// BoundParser is ArgumentParser bounded to an option, which provides
+/// API helpers to fill values in the bounded option.
+///
+/// Example:
+///         struct Options: BoundParserOption {
+///             var verbose = false
+///         }
+///         let parser = BoundParser<Options>(usage: "Sample", overview: "Sample")
+///         parser.add(option: "--verbose", kind: Bool.self) { $0.verbose = $1 }
+///         let options = try parser.parse(["--verbose"])
+public final class BoundParser<OptionType: BoundParserOption> {
```

Comment:
I would call this something better than `OptionType`, Option sounds like an individual option to me, despite the names we use in some other places.